File: src/FrontEnd.java
import spec.Language;
import spec.Library;

public class FrontEnd implements Programmer{
    private Language language;
    private Library library;

    @Override
    public Program makeProgram(Paper paper) {
        paper.setData(this);
        return makeFrontEndProgram();
    }

    private Program makeFrontEndProgram() {
        return new Program();
    }

    public void setLibrary(Library library) {
        this.library = library;
    }

    public void setLanguage(Language language) {
        this.language = language;
    }
}

File: Swift/Programmer/Programmer/Example/Programmer/FrontEnd.swift
//
//  ClientFrontEnd.swift
//  Programmer
//
//  Created by tskim on 19/10/2019.
//  Copyright © 2019 assin. All rights reserved.
//

import Foundation


protocol FrontEnd: Programmer {
    var language: Language? { get }
    var library: Library? { get }
}

extension FrontEnd {
    func makeProgram() -> Program {
        return Program()
    }
}
